### PR TITLE
Parse ical on external server

### PR DESCRIPTION
**Pull request by czacha338**
Hi, 

I have created an option to parse ical data on external server, it can be any server on which, the ical php scripts are placed. In my home I have a reference to locally created apache server. 

Example of config:
```
//parse ical on external server
config['calendar_parse_on_server'] = true;
config['calendar_parse_on_server_url'] = 'http://XXX.XXX.XXX.XXX/dashticz/ical'
```

Changed files:
- settings.js
- calendar.js
- en_US.json
- index.php

### CODE REVIEW

File: js/garbage.js [hunk 1]
```diff
@@ -380,7 +380,7 @@ function getEdgData(address, date, random) {
 
 function getTrashRow(garbage) {
     this.rowClass = 'trashrow';
-    this.displayDate = garbage.date.locale(settings['calendarlanguage']).format('l');
+    this.displayDate = garbage.date.locale(settings['calendarlanguage']).format(settings['calendarformat']);
```
Comment (czacha338) [inline js/garbage.js]: this cannot be merged

File: js/garbage.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ function loadGarbage() {
     if (typeof(settings['garbage_hideicon']) !== 'undefined' && parseFloat(settings['garbage_hideicon']) === 1) {
         html += '<div class="col-xs-12 col-data">';
     } else {
-        html += '<div class="col-xs-4 col-icon">';
+        html += '<div class="col-xs-4 col-icon-trash">';
```
Comment (czacha338) [inline js/garbage.js]: this cannot be merged

**Comment by czacha338** [comment]
The garbage.js file cannot be merged. There are some project specific adaptations.

**Comment by lokonli** [comment]
I just created something similar. See #391

**Comment by czacha338** [comment]
Ohhh... Sorry, I have not seen it :)

**Comment by lokonli** [comment]
Can you test whether the solution works for you? I did not test all situations ...

**Comment by czacha338** [comment]
So I have one error in site inspector (ical data are not parsed). 

> 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://XXX.XXX.XXX.XXX:8080' is therefore not allowed access.

So I would recommend to correct this by adding into index.php header:

```
<?php
header("Access-Control-Allow-Origin: *");
require_once('SG_iCal.php');
$ICS = $_GET['url'];
...
```

I have moved the ical part on some server and every device is parsing ical data through this server.

**Comment by lokonli** [comment]
Good point. It seems to work with the header. I'll create a PR.

**Comment by lokonli** [comment]
See #399.

**Pull request closed without merging**